packages/db-model/src/model.ts

```diff
@@ -72,86 +70,96 @@ export function createBaseModel(resolver?: ConnectionResolver) {
     }
 
     /**
-     * Generic query builder.
-     *
-     * @param {(qb: Knex | Transaction) => QueryBuilder} callback
-     * @param {Transaction} [trx]
-     * @returns {Promise<T[]>}
-     */
-    public static async buildQuery<T>(
-      callback: (qb: Knex | Transaction) => QueryBuilder,
-      trx?: Transaction
-    ): Promise<T[]> {
-      const qb = db.queryBuilder(this.getConnection(), trx);
-      const result = await callback(qb);
-
-      return object.toCamelCase<T[]>(result);
-    }
-
-    /**
-     * Finds a record based on the params.
-     * Returns null if no results were found.
+     * Finds a single record based on the params.
      *
      * @param {object} [params={}]
      * @param {Knex.Transaction} trx
-     * @returns {Promise<T | null>}
+     * @returns {Promise<T>}
      */
-    public static get<T>(params: object = {}, trx?: Knex.Transaction): Promise<T | null> {
-      return db.get<T>(this.getConnection(), this.table, params, trx);
-    }
+    public static findFirst<T>(params: object = {}, trx?: Knex.Transaction): Promise<T> {
+      return new Promise<T>(async (resolve, reject) => {
```

Comment:
Why wrap inside a promise? This won't work?

```suggestion
      return db.findFirst<T>(this.getConnection(), this.table, params, this.defaultOrderBy, trx);
```

Reply:
As I have made all the model functions async, the promise wrapper is added to extract "result" property from the Knex result. I believe this will result in cleaner code at the invocation side.